sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -38,7 +38,10 @@ def _validate_sale_orders(self, order_filter):
         _logger.debug('Sale Orders to validate: %s', sales.ids)
         for sale in sales:
             with savepoint(self.env.cr):
+                user_company = self.env.user.company_id
```

Comment:
What we would need is to avoid the 2 updates on user.company_id.

Could we have the user changed only in cache like we do in onchange context manager ?